development.md

```diff
@@ -0,0 +1,156 @@
+# Development
+
+## Introduction
+
+OpenRazer consists of a Linux kernel module and Python daemon + library. The Linux kernel module exposes an
+sysfs interface, the Python daemon (openrazer-daemon) exposes this on it's turn over D-Bus.
+
+If you want to add support for a new hardware device it's often required to make changes to both the kernel module
+as the Python sources. For example `io.github.openrazer.openrazer.metainfo.xml` and `99-razer.rules`. The
+fake drivers can be regenerated by running `./scripts/generate_all_fake_drivers.sh -f`.
+
+This description assumes you have cloned the openrazer repository to your home directory.
+
+## Setup your development environment
+
+These instructions should help you setup a development environment for both the kernel module and Python
+daemon.
+
+### Kernel module
+
+It is possible to do development and test your changes on your local system but you risk crashes and possible data
+loss. I really recommend to do your development on a separate virtual or physical machine.
```

Comment:
I've never really had a problem doing development on the host, which makes it way less complicated to set up. Of course mentioning that kernel modules can lead to crashes should be kept.

Reply:
I had very annoying crashes due to stupid bugs (while developing). Especially with `find_translation` and reading/writing to the null pointer...

I could often reboot but that really ruins your focus. I still really recommend a remote/virtual development machine.